graspy/subgraph/subgraph.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2019 NeuroData (http://neurodata.io)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from mgcpy.independence_tests.mgc import MGC
+from mgc.independence import Dcorr, RV, CCA
+
+
+def non_iterative_screen(a_tensor, y_labels, c, opt):
+    """
+    Performs non-iterative screening on graphs to estimate signal subgraph.
+    
+    Parameters
+    ----------
+    a_tensor: np.ndarray, shape (n_graphs, n_vertices, n_vertices)
+        Tensor of adjacency matrices
+    y_labels: np.ndarray, shape (n_graphs, 1)
+        Vector of ground truth labels
+    c: float
+        Correlation threshold value chosen by user
+    opt: string
+        Indication of which statistic to use
+    
+    Returns
+    -------
+    corrs: np.ndarray, shape (n_vertices, 1)
+        Vector of correlation values for each node
+    S_hat: np.ndarray, shape (ss_size, 1)
+        Estimated signal subgraph, approximated with non-iterative
+        vertex screening.
+        
+    References
+    ----------
+    .. [1] S. Wang, C. Chen, A. Badea, Priebe, C.E., Vogelstein, J.T.  "Signal 
+    Subgraph Estimation Via Vertex Screening," arXiv: 1801.07683 [stat.ME], 2018
+    """
+    if type(a_tensor) is not np.ndarray:
+        raise TypeError("a_tensor must be numpy.ndarray")
+    if a_tensor.shape[1] != a_tensor.shape[2]:
+        raise ValueError("Entries in a_tensor must be square matricies")
+    if len(a_tensor.shape) != 3:
+        raise ValueError("a_tensor must be a tensor")
+    # Import needed statistical modules from mgcpy package
+    mgc = MGC()
+    dcorr = Dcorr()
+    rv = RV()
+    cca = CCA()
+
+    # Finding dimension of each matrix
+    N = len(a_tensor[0])
+
+    # Create vector of zeros that will become vector of correlations
+    corrs = np.zeros((N, 1))
+
+    for i in range(N):
+
+        # Stacks the ith row of each matrix in tensor,
+        # creates matrix with dimension len(a_tensor) by N
+        mat = a_tensor[:, i]
+
+        # Statistical measurement chosen by the user
+        if opt == "mgc":
+            c_u_0, independence_test_metadata_0 = mgc.test_statistic(mat, y_labels)
+            corrs[i][0] = c_u_0
+        elif opt == "dcorr":
```

Comment:
these else ifs are unnecessary (besides the first) since you are calling _statistic on all of them